### PR TITLE
fix(logger): Ensure external loggers doesn't propagate logs

### PR DESCRIPTION
**Pull request by mploski**
**Issue #, if available:**

https://github.com/awslabs/aws-lambda-powertools-python/issues/1073

## Description of changes:

Ensure external loggers doesn't propagate logs to root logger by disabling Propagate attribute. Previously event was emitted in external logger ( with powertools handler) and in root handler IF propagation was set to True. This change ensure log is only emitted by external logger handler.

Also, during debugging noticed that we don't exclude `aws_lambda_powertools`  from handler modification since this is root powertools package we might treat it as internal package and skip applying powertools logger handler. @heitorlessa  please correct me if I'm wrong here 

**Checklist**

 

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] Update tests
* [ ] Update docs
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/376ec0a2ac0d2a40e0af5717bef42ff84ca0d1b9/.github/semantic.yml#L2)

## Breaking change checklist

**RFC issue #**:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by mploski** [comment]
@heitorlessa please review

**Review by heitorlessa — CHANGES_REQUESTED** [review]
Additional test to ensure we don't break child loggers

File: aws_lambda_powertools/logging/utils.py [hunk 1]
```diff
@@ -75,6 +75,7 @@ def _find_registered_loggers(
 def _configure_logger(source_logger: Logger, logger: logging.Logger, level: Union[int, str]) -> None:
     logger.handlers = []
     logger.setLevel(level)
+    logger.propagate = False
```
[the remark this answers is not in the document]
Comment (mploski) [reply aws_lambda_powertools/logging/utils.py:78]: Wasn't aware of this side-effect. Will create test case for it and work to ensure there is no regression. Thanks for spotting this :-)

[the remark this answers is not in the document]
Comment (mploski) [reply aws_lambda_powertools/logging/utils.py:78]: Wasn't aware of this side-effect. Will create test case for it and work to ensure there is no regression. Thanks for spotting this :-)

[the remark this answers is not in the document]
Comment (heitorlessa) [reply aws_lambda_powertools/logging/utils.py:78]: As I was running the tests locally and getting familiar with the code changes, I realized this wasn't true because we are filtering out child loggers by default -- copied config only applies to parent loggers so this edge case will never be hit.

I renamed the test covering keys being added after config is copied, so we can always come back to and know what this test is for.

> Repurposed test

```python
def test_copy_config_to_parent_loggers_only(stdout):
    # GIVEN Powertools Logger and Child Logger are initialized
    # and Powertools Logger config is copied over
    service = service_name()
    child = Logger(stream=stdout, service=service, child=True)
    parent = Logger(stream=stdout, service=service)
    utils.copy_config_to_registered_loggers(source_logger=parent)

    # WHEN either parent or child logger append keys
    child.append_keys(customer_id="value")
    parent.append_keys(user_id="value")
    parent.info("Logger message")
    child.info("Child logger message")

    # THEN both custom keys should be propagated bi-directionally in parent and child loggers
    # as child logger won't be touched when config is being copied
    parent_log, child_log = capture_multiple_logging_statements_output(stdout)
    assert "customer_id" in parent_log, child_log
    assert "user_id" in parent_log, child_log
    assert child.parent.name == service
```
> Where we ignore child loggers

```python
def _exclude_registered_loggers_filter(loggers: Set[str]) -> List[logging.Logger]:
    return [
        logging.getLogger(name) for name in logging.root.manager.loggerDict if "." not in name and name not in loggers
    ]
```

**Review by heitorlessa — APPROVED** [review]

**Comment by heitorlessa** [comment]
@mploski made a performance change (package_logger as constant over logger instance), removed an unnecessary test, and repurposed that child logger test to cover the rule of not touching child loggers.

I'll merge and make a release once you're okay with it

**Comment by heitorlessa** [comment]
In another PR, it's worth cleaning up the test functions/fixtures as I noticed they're duplicated from functional/test_logger - in this case, it's easier to have a logger utils for our tests and easily reuse via import (low priority)

**Comment by mploski** [comment]
> 

We need to also cover one blind spot to not break passing keys between child and parent. We may either prohibit from passing child logger as source to this method our automatically decapsulate parent logger and exclude it from modification to avoid setting propagate=False as it breaks things.

**Comment by mploski** [comment]
> @mploski made a performance change (package_logger as constant over logger instance), removed an unnecessary test, and repurposed that child logger test to cover the rule of not touching child loggers.
> 
> I'll merge and make a release once you're okay with it

We need to also cover one blind spot to not break passing keys between child and parent. We may either prohibit from passing child logger as source to this method our automatically decapsulate parent logger and exclude it from modification to avoid setting propagate=False as it breaks things.

**Comment by heitorlessa** [comment]
Awesome, checking as soon as I finish one last meeting
